processor/schemaprocessor/transformer.go

```diff
@@ -24,45 +23,120 @@ import (
-	conf config.Processor,
-	set component.ProcessorCreateSettings,
-) (*transformer, error) {
+func newTransformer(ctx context.Context, conf config.Processor, set component.ProcessorCreateSettings) (*transformer, error) {
 	cfg, ok := conf.(*Config)
 	if !ok {
 		return nil, errors.New("invalid configuration provided")
 	}
+
+	m, err := translation.NewManager(
+		cfg.Targets,
+		set.Logger.Named("schema-manager"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, prefetch := range cfg.Prefetch {
+		set.Logger.Info("Adding prefetch translation", zap.String("schema-url", prefetch))
+		_ = m.RequestTranslation(ctx, prefetch)
+	}
+
 	return &transformer{
-		log:     set.Logger,
-		targets: cfg.Targets,
+		config:    cfg,
+		telemetry: set.TelemetrySettings,
+		log:       set.Logger,
+		manager:   m,
 	}, nil
 }
 
 func (t transformer) processLogs(ctx context.Context, ld plog.Logs) (plog.Logs, error) {
+	ctx = context.NewOptimized(ctx)
+	for rl := 0; rl < ld.ResourceLogs().Len(); rl++ {
+		rLog := ld.ResourceLogs().At(rl)
+		t.manager.
+			RequestTranslation(ctx, rLog.SchemaUrl()).
+			ApplyAllResourceChanges(ctx, rLog)
+		for sl := 0; sl < rLog.ScopeLogs().Len(); sl++ {
+			log := rLog.ScopeLogs().At(sl)
+			t.manager.
+				RequestTranslation(ctx, log.SchemaUrl()).
+				ApplyScopeLogChanges(ctx, log)
+		}
+	}
 	return ld, nil
 }
 
 func (t transformer) processMetrics(ctx context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+	ctx = context.NewOptimized(ctx)
+	for rm := 0; rm < md.ResourceMetrics().Len(); rm++ {
+		rMetric := md.ResourceMetrics().At(rm)
+		t.manager.
+			RequestTranslation(ctx, rMetric.SchemaUrl()).
+			ApplyAllResourceChanges(ctx, rMetric)
+		for sm := 0; sm < rMetric.ScopeMetrics().Len(); sm++ {
+			metric := rMetric.ScopeMetrics().At(sm)
+			t.manager.
+				RequestTranslation(ctx, metric.SchemaUrl()).
+				ApplyScopeMetricChanges(ctx, metric)
+		}
+	}
 	return md, nil
 }
 
 func (t transformer) processTraces(ctx context.Context, td ptrace.Traces) (ptrace.Traces, error) {
+	ctx = context.NewOptimized(ctx)
+	for rt := 0; rt < td.ResourceSpans().Len(); rt++ {
+		rTrace := td.ResourceSpans().At(rt)
+		t.manager.
+			RequestTranslation(ctx, rTrace.SchemaUrl()).
+			ApplyAllResourceChanges(ctx, rTrace)
+		for ss := 0; ss < rTrace.ScopeSpans().Len(); ss++ {
+			span := rTrace.ScopeSpans().At(ss)
+			t.manager.
+				RequestTranslation(ctx, span.SchemaUrl()).
+				ApplyScopeSpanChanges(ctx, span)
+		}
+	}
 	return td, nil
 }
 
 // start will load the remote file definition if it isn't already cached
 // and resolve the schema translation file
 func (t *transformer) start(ctx context.Context, host component.Host) error {
-	for _, target := range t.targets {
-		t.log.Info("Fetching remote schema url", zap.String("schema-url", target))
+	var providers []translation.Provider
+	// Check for additional extensions that can be checked first before
+	// perfomring the http request
+	// TODO(MovieStoreGuy): Check for storage extensions
```

Comment:
Add in check if storage is being used, and pigging back on that and add write back feature for translations.

New PR for this part.